backend/src/controllers/user/index.ts

```diff
@@ -0,0 +1,54 @@
+import { Request, Response } from 'express';
+import { adminFireAuth } from '../../firebase';
+import { user } from '../../interfaces/user';
+import db from '../../config/db-config';
+
+export const getUserById = async (req: Request, res: Response) => {
+  try {
+    const uid = req.params.uid;
+    const user = await adminFireAuth.getUser(uid);
+    res.status(200).send({ user });
+  } catch (error) {
+    res.status(400).json(error);
+  }
+};
+
+export const updateUserById = async (req: Request, res: Response) => {
+  try {
+    const uid = req.params.uid;
+
+    const { first_name, last_name, photourl, phone_number } = req.body;
+    const user = await adminFireAuth.updateUser(uid, {
+      displayName: `${first_name},${last_name}`,
+      photoURL: photourl,
+      phoneNumber: phone_number,
+    });
+
+    const userInfo = await db('users')
+      .where('id', uid)
+      .update(first_name, last_name, photourl, phone_number);
+
+    res.status(200).send({ user, userInfo });
+  } catch (error) {
+    res.status(400).json(error);
+  }
```

Comment:
If firebase fails, it is not neccessarily because of the request being bad. I think the error handling here can be a bit better so that it is communicated correctly to the client what the issue is. Same comment for other places where you catch errors.

Reply:
Thank you very much for your review Kristian. I've made some changes to my code. But I am not 100% sure that I did exactly what you mean.